source/common/http/utility.cc

```diff
@@ -369,12 +370,23 @@ std::string Utility::makeSetCookieValue(const std::string& key, const std::strin
 }
 
 uint64_t Utility::getResponseStatus(const ResponseHeaderMap& headers) {
+  absl::StatusOr<uint64_t> response_status_or_absl_status = getResponseStatusOr(headers);
+  RELEASE_ASSERT(response_status_or_absl_status.ok(),
+                 ":status must be specified and a valid unsigned long");
+  return response_status_or_absl_status.value();
+}
+
+absl::StatusOr<uint64_t> Utility::getResponseStatusOr(const ResponseHeaderMap& headers) {
   const HeaderEntry* header = headers.Status();
+  absl::StatusOr<uint64_t> response_status_or_absl_status;
   uint64_t response_code;
   if (!header || !absl::SimpleAtoi(headers.getStatusValue(), &response_code)) {
-    throw CodecClientException(":status must be specified and a valid unsigned long");
+    response_status_or_absl_status =
+        absl::InvalidArgumentError(":status must be specified and a valid unsigned long");
+    return response_status_or_absl_status;
   }
-  return response_code;
+  response_status_or_absl_status = response_code;
+  return response_status_or_absl_status;
```

Comment:
Some of this seems redundant. Instead of declaring response_status_or_absl_status at the top, you should be able to omit it and just return the `absl::InvalidArgumentError` result or numeric response code, and rely on the implicit conversion constructors for `absl::StatusOr` to do the right thing.

Reply:
Yeah, good point. Changed.